### PR TITLE
8312547: Max/Min nodes Value implementation could be improved

### PR DESCRIPTION
**Pull request by merykitty**
Hi,

This patch removes the early return in `AddNode::Value` in case one of the inputs is a bottom, which may affect the value calculation of nodes such as `Min/MaxNode`.

Please kindly review, thanks very much.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8312547](https://bugs.openjdk.org/browse/JDK-8312547): Max/Min nodes Value implementation could be improved (**Enhancement** - P4)

### Reviewers
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**) ⚠️ Review applies to [ef2d3dfb](https://git.openjdk.org/jdk/pull/15021/files/ef2d3dfb5311c2567b31496e929b1506ea5cd5b7)
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15021/head:pull/15021` \
`$ git checkout pull/15021`

Update a local copy of the PR: \
`$ git checkout pull/15021` \
`$ git pull https://git.openjdk.org/jdk.git pull/15021/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15021`

View PR using the GUI difftool: \
`$ git pr show -t 15021`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15021.diff">https://git.openjdk.org/jdk/pull/15021.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15021#issuecomment-1650028204)

### CODE REVIEW

File: src/hotspot/share/opto/addnode.cpp
Comment (jatin-bhateja) [inline src/hotspot/share/opto/addnode.cpp:227]: We can  keep (t1 == Type::BOTTOM || t2 == Type::BOTTOM) condition intact.

Comment (merykitty) [reply src/hotspot/share/opto/addnode.cpp:227]: I don't think `Type::BOTTOM` can appear here, there are other nodes (such as convert nodes) that do not check for `Type::BOTTOM`. Maybe someone could shed light on this, please?

Comment (rwestrel) [reply src/hotspot/share/opto/addnode.cpp:227]: I don't think `Type::BOTTOM` is possible here either.

**Comment by merykitty** [comment]
@VladimirKempik Thanks for taking a look, the results look the same both before and after the patch, which is expected given none of them would benefit from more precise values of max/min nodes, these are the numbers:

                                            Before           After
    Benchmark                Mode  Cnt   Score   Error   Score   Error  Units
    MaxMinOptimizeTest.dAdd  avgt   12  59.865 ± 0.445  59.822 ± 0.323  us/op
    MaxMinOptimizeTest.dMax  avgt   12  61.109 ± 3.639  59.703 ± 0.187  us/op
    MaxMinOptimizeTest.dMin  avgt   12  59.961 ± 0.389  59.788 ± 0.281  us/op
    MaxMinOptimizeTest.dMul  avgt   12  59.477 ± 0.277  59.446 ± 0.337  us/op
    MaxMinOptimizeTest.fAdd  avgt   12  59.230 ± 0.315  59.103 ± 0.174  us/op
    MaxMinOptimizeTest.fMax  avgt   12  59.344 ± 0.422  60.171 ± 1.672  us/op
    MaxMinOptimizeTest.fMin  avgt   12  59.581 ± 0.606  59.187 ± 0.487  us/op
    MaxMinOptimizeTest.fMul  avgt   12  59.001 ± 0.236  58.948 ± 0.310  us/op

File: src/hotspot/share/opto/addnode.cpp
Comment (merykitty) [reply src/hotspot/share/opto/addnode.cpp:227]: The JVM is strongly typed so I don't think anything that is not higher than the local bottom can appear here. Note that the nodes will always return the local bottom or higher on `Value` invocation.

File: src/hotspot/share/opto/addnode.cpp
Comment (TobiHartmann) [inline src/hotspot/share/opto/addnode.cpp:220]:
```suggestion
  const Type* t1 = phase->type(in(1));
  const Type* t2 = phase->type(in(2));
```

**Review by TobiHartmann — APPROVED** [review]
Looks good to me.

**Comment by merykitty** [comment]
@TobiHartmann Thanks for your reviews, I have adjusted the code style there.

**Comment by merykitty** [comment]
Hi, may I have a second review, please?

**Review by vnkozlov — APPROVED** [review]
I am fine with fix and cleaning you did.

**Comment by merykitty** [comment]
@vnkozlov Thanks a lot for your reviews, I will integrate the change.

**Pull request closed without merging**
